backend/src/main/kotlin/org/darkSolace/muse/userModule/repository/UserRepository.kt

```diff
@@ -34,4 +34,20 @@ interface UserRepository : CrudRepository<User, Long> {
      * @return a [List] of [User]s having the required [UserTag]
      */
     fun findAllByUserTags(tag: UserTag): List<User>
+
+    /**
+     * Checks if a [User] with a given username already exists
+     *
+     * @param username The username to check
+     * @return [Boolean] weather the [User] already exists
+     */
+    fun existsByUsername(username: String): Boolean
```

Comment:
Please add a unit tests for this and the next function to check if the implementation is case sensitive.
existing username zero-udo and a call with Zero-Udo should return true